### PR TITLE
Changed the Dockerfile to use default toolset. Also improved Boost se…

### PR DESCRIPTION
**Pull request by dongxinEric**
…tup instructions in README.

### CODE REVIEW

**Comment by dongxinEric** [comment]
@apkar @AlvinMooreSr I did following tests against the changes:

- Build the Docker image from the new Dockerfile successfully
- Inside the aforementioned image, build Document Layer successfully using CMake.
- Inside the same container, run the Document Laye server and run full correctness tests against it successfully.

**Comment by dongxinEric** [comment]
This PR is meant to reduce confusions like #19

**Comment by apkar** [comment]
Test this please

**Comment by AlvinMooreSr** [comment]
test this please

**Comment by apkar** [comment]
test this please

**Comment by dongxinEric** [comment]
test this please

File: docker/Dockerfile [hunk 1]
```diff
@@ -7,9 +7,6 @@ RUN mkdir -p /opt && \
     curl -L -J https://dl.bintray.com/boostorg/release/1.67.0/source/boost_1_67_0.tar.gz | tar -xz && \
     cd boost_1_67_0 && \
     ./bootstrap.sh --prefix=./ &&\
-    CC=/usr/bin/gcc CXX=/usr/bin/g++ &&\
-    echo "using gcc : fdb : ${CXX} ;" >> ./tools/build/src/user-config.jam &&\
-    cat ./tools/build/src/user-config.jam &&\
-    ./b2 toolset=gcc-fdb install --with-filesystem --with-system
+    ./b2 install --with-filesystem --with-system
```
Comment (AlvinMooreSr) [inline docker/Dockerfile]: You should increment the LABEL version=0.0.1 line to 0.0.2

Comment (AlvinMooreSr) [reply docker/Dockerfile]: You should update the version number of the Dockerfile within the docker/docker-compose.yaml file to the new version included within the docker/Dockerfile

Comment (dongxinEric) [reply docker/Dockerfile]: dc7996e

**Comment by AlvinMooreSr** [comment]
test this please

**Comment by AlvinMooreSr** [comment]
test this please

**Comment by dongxinEric** [comment]
@AlvinMooreSr PR looks good now?

**Comment by AlvinMooreSr** [comment]
LGTM

File: docker/Dockerfile (hunk 1 above)
Comment (AlvinMooreSr) [reply docker/Dockerfile]: LGTM

**Pull request merged**